base/LineEdit.jl

```diff
@@ -725,6 +725,8 @@ function match_input(k::Dict, s, term=terminal(s), cs=Char[], keymap = k)
     # return an empty keymap function
     eof(term) && return keymap_fcn(nothing, "")
     c = read(term, Char)
+    # Ignore any '\0' (CTRL-space in xterm) - it's used by a special value elsewhere
```

Comment:
Good question.  It seems to be used as a special placeholder for the wildcard match of any character - in the keybinding syntax `'*'` is transformed into `'\0'` when normalizing a key binding:

https://github.com/JuliaLang/julia/blob/master/base/LineEdit.jl#L644
https://github.com/JuliaLang/julia/blob/master/base/LineEdit.jl#L650

The main point being that the LineEdit code assumes it's not possible to generate a `'\0'` from the terminal, but when it is possible, things break.